src/components/Application-Header/JobsAppBar.tsx

```diff
@@ -7,31 +10,42 @@ import Typography from '@material-ui/core/Typography';
 import Avatar from '@material-ui/core/Avatar';
 import Tooltip from '@material-ui/core/Tooltip';
 
-import { User } from '../../types/userTypes';
+import { User } from '../../types/User';
 import  jobsLogo from '../../assets/images/jobsLogo.png';
+import { UserStore } from '../../store/UserStore';
 import FavoriteList from './favorite-list/FavoriteList';
 import styles from './JobsAppBarStyle';
 
 interface AppBarDataProps extends WithStyles<typeof styles> {
-    user: User;
+    userStore?: UserStore;
 }
 
 const JobsAppBar: React.FC<AppBarDataProps> = (props): JSX.Element => {
-    const { user, classes } = props;
+    const { classes } = props;
+
+    const userStore: UserStore = props.userStore!;
+    const user: User = userStore.getUser;
+
+    useEffect(() => {
+        userStore.loadUserDetails();
+    }, []);
 
-    // For now I used a stupid function, just to get a different color   
-    const userHaveFavorites = (): boolean => {
-        return Math.random() >= 0.5;
+    const getUserInitials = (name: string): string => {
+        const splitName: string[] = name.split(' ');
+        return splitName.map(n=> n[0]).join(' '); 
     }
 
-    const getUserDetails = (): JSX.Element => {
+    const getUserDetails = (): JSX.Element | void  => {
+        if (userStore.isLoading || user === undefined) {
+            return;
+        }
         return (
             <div className={classes.userDetails}>
                 <Tooltip placement="right" title="ג'ובים ששמרתי" aria-label="my favorites">
                     <FavoriteList/>
                 </Tooltip>
                 <Avatar className={classes.avatar}>
-                    {user.userInitials}
+                    {getUserInitials(user.name)}
```

Comment:
Why not store method in `mobx`? it is used globally in project 😃

Reply:
Done